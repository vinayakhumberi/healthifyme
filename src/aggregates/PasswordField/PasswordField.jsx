import React, { useState, useEffect, useRef } from "react";
import PropTypes from "prop-types";
import { Container, MessageSpace } from "./style";
import Text from '../../components/Text/Text';
import TextInput from '../../components/TextInput/TextInput';
import {
  PASSWORD_REGEX
} from '../../constants/general.constants';

const PasswordField = function (props) {
  const [showMessage, setShowMessage] = useState('');
  const [valid, setValid] = useState(false);
  const passwordRef = useRef('');
  useEffect(() => {
    props.updatePassword({
      value: passwordRef.current,
      status: valid
    })
  }, [valid]);
  const handleTextChange = (e) => {
    const data = e.target.value;
    if (showMessage) setShowMessage("");
    if (valid) setValid(false);
    passwordRef.current = data;
    verifyPassword(data, false);
  };
  const handleVerify = (e) => {
    const data = e.target.value;
    verifyPassword(data, true);
  };
  const verifyPassword = (data, error) => {
    if (data.length === 0 && error) {
      setShowMessage("Password is mandatory");
    } else if (data.length > 0 && data.length <= 6 && error) {
      setShowMessage("Should me more than 6 characters long");
    } else {
      if (PASSWORD_REGEX.test(data)) {
        setValid(true);
      } else {
        if (error) setShowMessage("Should contain 1 uppercase letter");
      }
    }
  };
  return (
    <Container>
      <TextInput
        elementType="password"
        marginTop={10}
        marginBottom={0}
        paddingTop={10}
        paddingBottom={10}
        paddingLeft={10}
        paddingRight={10}
        width="100%"
        placeHolder="Enter your Password"
        borderColor={showMessage ? '#ec445c' : '#eee'}
        onChange={handleTextChange}
        onBlur={handleVerify}
      />
      <MessageSpace>
        {showMessage && <Text
          text={showMessage}
          color="#ec445c"
          paddingTop={5}
          paddingLeft={5}
        />}
      </MessageSpace>
    </Container>
  );
};
PasswordField.propTypes = {
  updatePassword: PropTypes.func
};

PasswordField.defaultProps = {
  updatePassword: () => {}
};

export default PasswordField;
