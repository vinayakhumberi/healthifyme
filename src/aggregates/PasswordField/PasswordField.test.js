import React from 'react';
import { mount, shallow, render } from 'enzyme';
import PasswordField from './PasswordField';

describe('<PasswordField />', () => {
  let wrapper;
  const updatePassword = () => { };
  beforeEach(() => wrapper = shallow(<PasswordField updatePassword={updatePassword} />));

  it('Should Render a PasswordField Container', () => {
    expect(wrapper.find('[data-test="password-field-container"]').length).toEqual(1);
  });
});


describe('<PasswordField /> error check', () => {
  let wrapper;
  const updatePassword = () => { };
  beforeEach(() => wrapper = mount(<PasswordField updatePassword={updatePassword} />));
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  // it('Should Render a PasswordField Container', () => {
  //   expect(wrapper.find('[data-test="password-field-container"]').length).toEqual(1);
  // });
  it('Should pass updatePassword prop to TextInput', () => {
    const input = wrapper.find('input');
    input.simulate('focus');
    input.simulate('change', { target: { value: 'Test Data' } });
    // expect(input.get(0).value).toEqual('Test Data');
  });
  it('Should show Error message is text is not more than 6 character long', () => {

  });
  it('Should show Error message if onBlur is fired and text is empty', () => {

  });
  it('Should show Error message if it is an invalid Password', () => {

  });
});
