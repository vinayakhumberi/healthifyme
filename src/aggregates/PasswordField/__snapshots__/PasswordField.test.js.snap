// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordField /> error check renders correctly 1`] = `ReactWrapper {}`;
